src/main/java/com/powsybl/openloadflow/sensi/AbstractSensitivityAnalysis.java

```diff
@@ -623,33 +661,55 @@ protected void fillRhsSensitivityVariable(SensitivityFactorGroupList<V, E> facto
+                                        Set<LfBranch> disabledBranches, PropagatedContingency propagatedContingency) {
         for (LfSensitivityFactor<V, E> factor : lfFactors) {
-            if (factor.getStatus() == LfSensitivityFactor.Status.VALID) {
-                // after a contingency, we check if the factor function and the variable are in different connected components
-                boolean variableConnected = factor.isVariableConnectedToSlackComponent(disabledBuses, disabledBranches);
-                boolean functionConnected = factor.isFunctionConnectedToSlackComponent(disabledBuses, disabledBranches);
-                if (!variableConnected && functionConnected) {
-                    // VALID_ONLY_FOR_FUNCTION status
-                    factor.setSensitivityValuePredefinedResult(0d);
-                } else if (!variableConnected && !functionConnected) {
-                    // SKIP status
-                    factor.setSensitivityValuePredefinedResult(Double.NaN);
-                    factor.setFunctionPredefinedResult(Double.NaN);
-                } else if (variableConnected && !functionConnected) {
+            Pair<Optional<Double>, Optional<Double>> predefinedResults = getPredefinedResults(factor, disabledBuses, disabledBranches, propagatedContingency);
+            Optional<Double> sensitivityValuePredefinedResult = predefinedResults.getLeft();
+            Optional<Double> functionPredefinedResults = predefinedResults.getRight();
+            if (sensitivityValuePredefinedResult.isPresent()) {
+                factor.setSensitivityValuePredefinedResult(sensitivityValuePredefinedResult.get());
+            }
+            if (functionPredefinedResults.isPresent()) {
+                factor.setFunctionPredefinedResult(functionPredefinedResults.get());
+            }
+        }
+    }
+
+    protected Pair<Optional<Double>, Optional<Double>> getPredefinedResults(LfSensitivityFactor<V, E> factor, Set<LfBus> disabledBuses,
+                                                                            Set<LfBranch> disabledBranches, PropagatedContingency propagatedContingency) {
+        Double sensitivityValuePredefinedResult = null;
+        Double functionPredefinedResult = null;
+        if (factor.getStatus() == LfSensitivityFactor.Status.VALID) {
+            // after a contingency, we check if the factor function and the variable are in different connected components
+            // or if the variable is in contingency. Note that a branch in contingency is considered as not connected to the slack component.
+            boolean variableConnected = factor.isVariableConnectedToSlackComponent(disabledBuses, disabledBranches) && !factor.isVariableInContingency(propagatedContingency);
+            boolean functionConnectedToSlackComponent = factor.isFunctionConnectedToSlackComponent(disabledBuses, disabledBranches);
```

Comment:
Maybe I am wrong but for the function, we only have branches (`BRANCH_ACTIVE_POWER_1` for example) or buses (`BUS_VOLATGE`). For the variable, it is more complexe because we can have all kind of injections.